pkg/vm/stackitem/json.go

```diff
@@ -213,18 +213,35 @@ func (d *decoder) decode() (Item, error) {
 		return NewByteArray([]byte(t)), nil
 	case json.Number:
 		ts := t.String()
-		dot := strings.IndexByte(ts, '.')
-		if dot != -1 {
-			// As a special case numbers like 123.000 are allowed (SetString rejects them).
-			// And yes, that's the way C# code works also.
-			for _, r := range ts[dot+1:] {
-				if r != '0' {
-					return nil, fmt.Errorf("%w (real value for int)", ErrInvalidValue)
+		var (
+			num *big.Int
+			ok  bool
+		)
+		isScientific := strings.Contains(ts, "e+") || strings.Contains(ts, "E+")
+		if isScientific {
+			// As a special case numbers like 2.8e+22 are allowed (SetString rejects them).
+			// That's the way how C# code works.
+			f, _, err := big.ParseFloat(ts, 10, big.MaxPrec, big.ToNearestEven)
```

Comment:
We have a 256-bit VM, it should be sufficient instead of `big.MaxPrec`.

Reply:
Temporary set to `53` to match the ref. implementation. Testnet is OK with that, mainnet is being checked.